moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
I didn't think of it as a "solution to this problem". More as "this problem arises because the previous change that made `geometric_shapes` use c++11 shared_ptr was inconsistent.
And if we want to switch MoveIt! to std::shared_ptr (which is what this request is about), this should imho include geometric_shapes. Incidentally, I assumed *all* shared_ptr references where switched to c++11 when we discussed the issue with FCL and octomap.